program-runtime/src/invoke_context.rs

```diff
@@ -108,6 +108,67 @@ impl ComputeMeter {
     }
 }
 
+// bprumo TODO: doc where 128 GB came from. Move where the const is, also
+// 128 GB
+pub const DEFAULT_MAX_ACCOUNTS_DATA_LEN: usize = 128_000_000_000;
+
+/// bprumo TODO: doc, and/or move
+#[derive(Debug, Default, Clone, Copy, Eq, PartialEq)]
+pub struct AccountsDataBudget {
+    /// The global maximum size for accounts data (i.e. DEFAULT_MAX_ACCOUNTS_DATA_LEN)
+    maximum: usize,
+    /// The current accounts data size (for a Bank)
+    current: usize,
+}
+impl AccountsDataBudget {
+    pub fn new(maximum: usize, current: usize) -> Self {
```

Comment:
Is maximum something that is going to change dynamically?  If not could encapsulate that in `AccountDataBudget`

Reply:
I'm not sure if the maximum will change dynamically yet. This is still to make prototyping easiest, and to allow tests to set a different maximum as well. I agree that the max could be in `AccountsDataBudget` directly.